Samples/WindowsCopilotRuntime/Contributing.md

```diff
@@ -0,0 +1,92 @@
+# Contributing
+
+This shows the project structure and how one can get started contributing to Windows Copilot Runtime (WCR) sample.
+
+## Project Structure
+
+The project follows the MVVM (Model-View-ViewModel) pattern. Here's how the code is organized:
+
+- **Models**: Contains the data models used in the application.
+  - `cs-winui/Models/ImageDescriptionModel.cs`
+  - `cs-winui/Models/ImageObjectExtractorModel.cs`
+- **ViewModels**: Contains the view models that handle the business logic and data binding for the views.
+  - `cs-winui/ViewModels/ImageDescriptionViewModel.cs`
+  - `cs-winui/ViewModels/mageObjectExtractorViewModel.cs`
+- **Pages**: Contains the XAML pages and their code-behind files. They represent the `Views` part of MVVM.
+  - `cs-winui/Pages/ImageDescriptionPage.xaml`
+  - `cs-winui/Pages/ImageDescriptionPage.xaml.cs`
+  - `cs-winui/Pages/ImageObjectExtractorPage.xaml`
+  - `cs-winui/Pages/ImageObjectExtractorPage.xaml.cs`
+- **Helpers**: Util, Converters and Controls folders contain common shared code used by all the pages
+  - `Controls` folder contains user controls for showing creating Session and example source codes
+  - `CodeBlockControl.xaml` and `CodeBlockControl.xaml.cs` are parts of user control which shows example source code for each api. The example code is fetched from `Examples` folder.
+  - `ModelInitializationControl.xaml` and `ModelInitializationControl.xaml.cs` have user control to create session for every page. Each page sets `CreateModelSessionWithProgress` as `ICommand` to button in this control. 
+  
+## Adding a New Page
+Any of the examples like `ImageScaler` or `ImageDescriptionModel` can be as reference for adding newer pages. `LanguageModel` is slightly different as it has multiple effects in the sample page. 
+
+
+## AsyncCommand and Its Inherited Class
+
+In this project, we use `AsyncCommand` to handle asynchronous operations in the view models. `AsyncCommand` is a custom implementation of `ICommand` that supports asynchronous execution.
+
+### AsyncCommand
+
+Here's an example of how to use `AsyncCommand`:
+
+```csharp
+...
+    public ICommand ScaleCommand => _scaleSoftwareBitmapCommand;
+    private readonly AsyncCommand<(SoftwareBitmap, int, int), SoftwareBitmapSource> _scaleSoftwareBitmapCommand;
```

Comment:
Does the member `_scaleSoftwareBitmapCommand` need to exist? Would something like this work?

```csharp
public ICommand ScaleCommand { get; } = new AsyncCommand(async _ => { ... });
```

Or, if this can't be initialized directly inline with the property, the ctor could set it:

```csharp
public ICommand ScaleCommand { get; private set; }

public ImageScalerViewModel(ImageScalerModel imageScalerSession)
: base(imageScalerSession)
{
    ScaleCommand = new AsyncCommand(/*...*/);
}
```
```

Reply:
It can. I am directly referencing an already existing file's code so not doing any changes from that.  That would easier for user to follow